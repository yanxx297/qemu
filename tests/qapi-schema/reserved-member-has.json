# C member name collision
# FIXME - This parses, but fails to compile, because the C struct is given
# two 'has_a' members, one from the flag for optional 'a', and the other
# from member 'has-a'.  Either reject this at parse time, or munge the C
# names to avoid the collision.
{ 'command': 'oops', 'data': { '*a': 'str', 'has-a': 'str' } }
