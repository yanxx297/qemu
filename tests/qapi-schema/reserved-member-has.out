object :empty
object :obj-oops-arg
    member a: str optional=True
    member has-a: str optional=False
command oops :obj-oops-arg -> None
   gen=True success_response=True
